File: rb/features.rb
# CrossrefRest SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module CrossrefRestFeatures
  def self.make_feature(name)
    case name
    when "base"
      CrossrefRestBaseFeature.new
    when "ratelimit"
      CrossrefRestRatelimitFeature.new
    when "retry"
      CrossrefRestRetryFeature.new
    when "test"
      CrossrefRestTestFeature.new
    when "timeout"
      CrossrefRestTimeoutFeature.new
    else
      CrossrefRestBaseFeature.new
    end
  end
end

File: py/features.py
# CrossrefRest SDK feature factory

from feature.base_feature import CrossrefRestBaseFeature
from feature.test_feature import CrossrefRestTestFeature


def _make_feature(name):
    features = {
        "base": lambda: CrossrefRestBaseFeature(),
        "test": lambda: CrossrefRestTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
